### PR TITLE
Start service with start foregroundService on Android Oreo

### LINKED ISSUE #1785: StartService in Android Oreo and More should use startForegroundService
### Versions

* Python: 2.7
* OS: Android (API 28)
* Kivy: master
* Cython: recommendedone

### Description

On Android Oreo and more, Start service should use startForegroundService instead of startService

### PR DESCRIPTION
**Pull request by brvier**
And add a notification channel for Oreo permanent service notification
Fix #1785

### CODE REVIEW

**Comment by Ravi2232** [comment]
Yes we are

**Comment by inclement** [comment]
Thanks @brvier, the changes look basically good to me. It looks like travis is complaining about a build issue though, is this a real problem with the code?

File: pythonforandroid/bootstraps/common/build/templates/Service.tmpl.java [hunk 1]
```diff
@@ -11,19 +11,34 @@
...
 
 public class Service{{ name|capitalize }} extends PythonService {
+
+    private boolean autoRestartService = true;
+
     {% if sticky %}
     @Override
     public int startType() {
-        return START_STICKY;
+        return START_NOT_STICKY;
```
Comment (AndreMiras) [inline pythonforandroid/bootstraps/common/build/templates/Service.tmpl.java]: I'm really not getting that part :thinking: we have the `if sticky` in the template, but we decide to return not sticky.
There's something fishy about it no?

**Comment by AndreMiras** [comment]
@inclement  @brvier I think usually the CI fails for good reasons. Here for instance we have:
```
:compileDebugJavaWithJavac/home/user/.local/share/python-for-android/dists/bdisttest_python3_sqlite_openssl_googlendk/src/main/java/org/kivy/android/PythonService.java:22: error: package android.support.v4.app does not exist
import android.support.v4.app.ActivityCompat;
                             ^
/home/user/.local/share/python-for-android/dists/bdisttest_python3_sqlite_openssl_googlendk/src/main/java/org/kivy/android/PythonService.java:28: error: package android.support.v4.app does not exist
import android.support.v4.app.NotificationCompat;
                             ^
/home/user/.local/share/python-for-android/dists/bdisttest_python3_sqlite_openssl_googlendk/src/main/java/org/kivy/android/PythonService.java:139: error: package NotificationCompat does not exist
            NotificationCompat.Builder notificationBuilder = new NotificationCompat.Builder(context, "persistent");
                              ^
/home/user/.local/share/python-for-android/dists/bdisttest_python3_sqlite_openssl_googlendk/src/main/java/org/kivy/android/PythonService.java:139: error: package NotificationCompat does not exist
            NotificationCompat.Builder notificationBuilder = new NotificationCompat.Builder(context, "persistent");
                                                                                   ^
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
4 errors
 FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileDebugJavaWithJavac'.
> Compilation failed; see the compiler error output for details.
```

File: pythonforandroid/bootstraps/common/build/templates/Service.tmpl.java [hunk 2]
```diff
@@ -67,7 +102,26 @@ static public void start(Context ctx, String pythonServiceArgument) {
...
+
+
+    @Override
+    public void onDestroy() {
+        super.onDestroy();
+    }
+
+    /**
+     * Stops the task gracefully when killed.
+     * Calling stopSelf() will trigger a onDestroy() call from the system.
```
Comment (AndreMiras) [inline pythonforandroid/bootstraps/common/build/templates/Service.tmpl.java]: Misleading comment, also why do we override method if we don't add/change anything?

Comment (brvier) [reply pythonforandroid/bootstraps/common/build/templates/Service.tmpl.java]: Good point :) 
:+1:

**Comment by brvier** [comment]
>  I think usually the CI fails for good reasons.

Hopefully, else it s useless. 

So here Travis point to dependancies i have, and which isn't default on pythonforandroid.

Is there any problem to add this deps in pythonforandroid ?

**Comment by AndreMiras** [comment]
Yes if your pull request requires a dependency, then yes add that dep as part of your pull request and also think about about updating the docs in that same PR

**Comment by AndreMiras** [comment]
Up @brvier :wave: 
There's an up to date PR also here https://github.com/kivy/python-for-android/pull/1725 can you give it a try before we merge it?

**Comment by JonasT** [comment]
Is this superseded now and possibly should be closed? @inclement & @AndreMiras I'm seeing similar code already in current `develop`

**Comment by JonasT** [comment]
Since we seem to have something similar on `develop` already I'll close this for now. **Thanks so much for your contribution,** sorry it didn't make it into the project in exactly your provided form!

If anyone thinks this is closed by mistake let me know and I'll gladly immediately reopen

**Pull request closed without merging**